### PR TITLE
Add custom errors and improve loading states

### PR DESCRIPTION
**Pull request by AlexanderKolberg**

### CODE REVIEW

File: playgrounds/react-vite/src/tabs/Debug.tsx
Comment (w1that) [inline playgrounds/react-vite/src/tabs/Debug.tsx:1]: There are some missing imports and brackets for this file.

**Review by w1that — CHANGES_REQUESTED** [review]
Apart from this comment, looks great, thanks!

File: playgrounds/react-vite/src/tabs/Debug.tsx
Comment (AlexanderKolberg) [reply playgrounds/react-vite/src/tabs/Debug.tsx:1]: Thanks, fixed. Sometimes I hate rebasing...

File: playgrounds/react-vite/src/tabs/Debug.tsx [hunk 1]
```diff
@@ -1,270 +1,287 @@
...
 
-        <Box justifyContent="space-between">
-          <Text variant="large">Debug Tools</Text>
+	const getFunctionSignatures = () => {
+		const abi = ABIs[selectedAbi];
+		return abi
+			.filter((item) => item.type === 'function')
+			.map((func) => {
+				try {
+					const signature = toFunctionSelector(func);
```
Comment (w1that) [inline playgrounds/react-vite/src/tabs/Debug.tsx:74]:
```suggestion
					const signature = toFunctionSelector(func as AbiFunction);
```

File: playgrounds/react-vite/src/tabs/Debug.tsx [hunk 2]
```diff
@@ -1,270 +1,287 @@
...
+					return `${func.name}: ${signature}`;
```
Comment (w1that) [inline playgrounds/react-vite/src/tabs/Debug.tsx:75]:
```suggestion
					return `${func.type}: ${signature}`;
```

`func` doesn't seem to have a `name`.

**Review by w1that — APPROVED** [review]
LGTM

**Pull request merged**